source/development.rst

```diff
@@ -0,0 +1,139 @@
+===========
+Development
+===========
+It is recommended that PCDS Python projects are updated and maintained in the
+following way. It is assumed that all PCDS projects are maintained on either
+the SLAC Github or the PCDS specific site. This workflow was not invented here
+at SLAC, there are many helpful tutorials online like `this
+<https://guides.github.com/introduction/flow>`_ if you want more information.
+
+Creating a Local Checkout
+=========================
+If you want to make changes to a repository the first step is to create your
+own fork. This allows you to create feature branches without cluttering the
+main repository. It also assures that the main repository is only added to by
+Pull Request and review. Repositories can be forked from the GitHub site like
+`this <https://help.github.com/articles/fork-a-repo>`_. Once this repository is
+created, you can clone into your own workspace.
+
+.. code:: sh
+
+   $ git clone https://github.com/YOUR-USERNAME/REPOSITORY.git
+
+Now, that we have a copy of the repository create a branch for the feature or
+bug you would like to work on.
+
+.. code:: sh
+
+   $ git checkout -b my-feature
+
+   $ git status
+   On branch my-feature
+   nothing to commit, working tree clean
+
+Commit Guidelines
+=================
+Now you are ready to start working! Make changes to files and commit them to
+your new branch. We like to preface our commit messages with a descriptor code.
+This makes it easier for someone reviewing your commit history to see what you
+have done.  These are borrowed from the `NumPy
+<https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message>`_
+development documentation.
+
+====  ===
+Code  Description
+====  ===
+API   an (incompatible) API change
+BLD   change related to building
+BUG   bug fix
+DEP   deprecate something, or remove a deprecated object
+DEV   development tool or utility
+DOC   documentation
+ENH   enhancement
+MNT   maintenance commit (refactoring, typos, etc.)
+REV   revert an earlier commit
+STY   style fix (whitespace, PEP8)
+TST   addition or modification of tests
+REL   related to releasing numpy
+WIP   Commit that is a work in progress
+====  ===
+
+It is also helpful underneath classes and functions to write docstrings. These
+are later converted by Sphinx into HTML documentation. They also are a valuable
+tool for exploration of a codebase within an IPython terminal. Docstrings
+should follow the form described in the `NumPy documentation
+<http://www.sphinx-doc.org/en/stable/ext/example_numpy.html>`_
+
+Merging Changes
```

Comment:
There isn't a section on starting new repositories so 'Merging Changes' looks like the best section for mentioning it for now.